include/lemlib/PID.hpp

```diff
@@ -0,0 +1,123 @@
+#pragma once
+
+#include "units/units.hpp"
+#include <optional>
+
+namespace lemlib {
+/**
+ * @brief Struct to hold PID gains.
+ *
+ * @param kP proportional gain
+ * @param kI integral gain
+ * @param kD derivative gain
+ */
+struct Gains {
+        double kP = 0;
+        double kI = 0;
+        double kD = 0;
+};
+
+class PID {
+    public:
+        /**
+         * @brief Constructs a new PID controller
+         *
+         * @param kP proportional gain
+         * @param kI integral gain
+         * @param kD derivative gain
+         * @param windupRange range at which integral is reset
+         * @param signFlipReset whether to reset integral when error changes sign
+         */
+        PID(double kP, double kI, double kD, double windupRange = 0, bool signFlipReset = false);
+        /**
+         * @brief Constructs a new PID controller
+         *
+         * @param gains the gains to use
+         * @param windupRange range at which integral is reset
+         * @param signFlipReset whether to reset integral when error changes sign
+         */
+        PID(Gains gains, double windupRange = 0, bool signFlipReset = false);
```

Comment:
take a const reference to gains
```suggestion
        PID(const Gains& gains, double windupRange = 0, bool signFlipReset = false);
```